nixos/modules/services/hardware/asusctl.nix

```diff
@@ -0,0 +1,482 @@
+          scale = 0.9;
+          angle = 0.65;
+          translation = [
+            0.0
+            0.0
+          ];
+          time = {
+            Fade = {
+              fade_in = {
+                secs = 2;
+                nanos = 0;
+              };
+              show_for = {
+                secs = 2;
+                nanos = 0;
+              };
+              fade_out = {
+                secs = 2;
+                nanos = 0;
+              };
+            };
+          };
+          brightness = 1.0;
+        };
+      }
+    ];
+    shutdown = [
+      {
+        ImageAnimation = {
+          file = "${cfg.package}/usr/share/asusd/anime/custom/sonic-wait.gif";
+          scale = 0.9;
+          angle = 0.0;
+          translation = [
+            3.0
+            2.0
+          ];
+          time = "Infinite";
+          brightness = 1.0;
+        };
+      }
+    ];
+    brightness = 1.0;
+    awake_enabled = true;
+    boot_anim_enabled = true;
+  };
+
+  mkAnimeConfig = generators.toJSON { } (recursiveUpdate localAnimeConfig cfg.animeConfig);
+  animeConfigFile = pkgs.writeText "anime.conf" (mkAnimeConfig);
+
+  mkAsusdConfig = generators.toJSON { } (recursiveUpdate localAsusdConfig cfg.asusdConfig);
+  asusdConfigFile = pkgs.writeText "asusd.conf" (mkAsusdConfig);
+
+  mkAuraConfig = generators.toJSON { } (recursiveUpdate localAuraConfig cfg.auraConfig);
+  auraConfigFile = pkgs.writeText "aura.conf" (mkAuraConfig);
+
+  profileConfigFile = pkgs.writeText "profile.conf" (cfg.profileConfig);
+  ledModesConfigFile = pkgs.writeText "asusd-ledmodes.conf" (cfg.ledModesConfig);
+in
+{
+  options = {
+
+    services.asusctl = {
+
+      enable = mkEnableOption (lib.mdDoc "Enable the asusctl service");
+
+      enableUserService = mkOption {
+        type = types.bool;
+        description = lib.mdDoc ''
+          Activate the asusd-user service.
+        '';
+        default = false;
+      };
+
+      animeConfig = mkOption {
+        type = types.attrs;
+        description = lib.mdDoc ''
+          The content of /etc/asusd/anime.conf.
+        '';
+        default = { };
+      };
+
+      asusdConfig = mkOption {
+        type = types.attrs;
+        description = lib.mdDoc ''
+          The content of /etc/asusd/asusd.conf.
+        '';
+        default = { };
+      };
+
+      auraConfig = mkOption {
+        type = types.attrs;
+        description = lib.mdDoc ''
+          The content of /etc/asusd/aura.conf.
+        '';
+        default = { };
+      };
+
+      profileConfig = mkOption {
+        type = types.str;
+        description = lib.mdDoc ''
+          The content of /etc/asusd/profile.conf.
+        '';
+        default = '''';
+      };
+
+      ledModesConfig = mkOption {
+        type = types.str;
+        description = lib.mdDoc ''
+          The content of /etc/asusd/asusd-ledmodes.toml.
+        '';
+        default = ''
+          [[led_data]]
```

Comment:
This default is way way to long and also If I would want to append one thing I would need to copy this all.